pilot/pkg/proxy/envoy/v2/debug.go

```diff
@@ -454,18 +462,20 @@ func (sd *MemServiceDiscovery) GetIstioServiceAccounts(hostname model.Hostname,
 func (s *DiscoveryServer) registryz(w http.ResponseWriter, req *http.Request) {
 	_ = req.ParseForm()
 	w.Header().Add("Content-Type", "application/json")
-	svcName := req.Form.Get("svc")
-	if svcName != "" {
-		data, err := ioutil.ReadAll(req.Body)
-		if err != nil {
-			return
-		}
-		svc := &model.Service{}
-		err = json.Unmarshal(data, svc)
-		if err != nil {
-			return
+	if debugMode {
```

Comment:
How about - just remove this section completely, no flags or anything.

Will add it back if needed - there are better options now for doing this kind of testing ( with a dummy MCP server )